dags/operators/emr_spark_operator.py

```diff
@@ -246,12 +246,13 @@ def describe_cluster():
                 spark_log_location = self.get_spark_log_location()
 
                 raise AirflowException(
-                    'Spark job {} terminated with errors: {} - {}<br>'
+                    u'Spark job {} terminated with errors: {} - {}<br>'
                     'Cluster Stderr: {}<br>'
                     'Cluster Stdout: {}<br>'
                     'Spark Driver Log Location: {}'
                     .format(self.job_name, reason_code, reason_message,
                             step_logs['stderr'], step_logs['stdout'], spark_log_location)
+                    .format('ascii', 'replace')
```

Comment:
Should this be `encode` instead? I tried the following in a 2.7 session:

```python
>>> u'hello world {}'.format(b"\xF0\x9F\x98\x81".decode('utf')).format('ascii', 'replace')
u'hello world \U0001f601'
>>> u'hello world {}'.format(b"\xF0\x9F\x98\x81".decode('utf')).encode('ascii', 'replace')
'hello world ?'
```

Reply:
And this, my friend, is why we have code review (and why we should have a testable operator but..)